Sources/Clients/GatewayAPI/GatewayAPIClient+Extension.swift

```diff
@@ -207,37 +207,58 @@ extension GatewayAPI.EntityMetadataCollection {
 extension GatewayAPI.ComponentEntityRoleAssignments {
 	// FIXME: This logic should not be here, will probably move to OnLedgerEntitiesClient.
 	@Sendable public func extractBehaviors() -> [AssetBehavior] {
```

Comment:
this becoming more and more complex, could benefit from  adding some tests

Reply:
Yes, there should definitely be tests, and I will add a longer comment explaining the logic